lib/msal-common/src/authority/Authority.ts

```diff
@@ -220,13 +228,24 @@ export class Authority {
      */
     public get jwksUri(): string {
         if(this.discoveryComplete()) {
-            const endpoint = this.replacePath(this.metadata.jwks_uri);
-            return this.replaceTenant(endpoint);
+            return this.replacePath(this.metadata.jwks_uri);
         } else {
             throw ClientAuthError.createEndpointDiscoveryIncompleteError("Discovery incomplete.");
         }
     }
 
+    /**
+     * Returns a flag indicating that authority {@link IUri} type is AAD
+     * @param authorityUri {@link IUri}
+     * @private
+     */
+    private isAADAuthority(authorityUri: IUri): boolean {
+        return authorityUri.PathSegments.length === 1
+            && !Authority.reservedTenantDomains.has(authorityUri.PathSegments[0])
```

Comment:
An authority with these reservedDomains could still be (is probably) an AAD authority. If you still need to do this check, can we do it at the call-site? Alternatively, we could rename the method to something more accurate. Something like `canReplaceTenant` perhaps?